### ISSUE TITLE
Compatibility between NewType(int) and float

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#2539]
Consider this code:
```python
from typing import NewType

Utime = NewType("Utime", int)

def f(x: Utime) -> float:
    return x
```
Pyright errors on the return line saying that `Utime` is incompatible with `float`. But a Utime is an int, and ints can be assigned to float following PEP 484, so my understanding is that this code should be allowed.

Mypy allows this too: https://mypy-play.net/?mypy=latest&python=3.10&gist=25769157cf1e66cdcac78ee69682cc80

**VS Code extension or command-line**
Pylance v2021.11.0 in VSCode

**Comment by erictraut** [comment]
Thanks for the bug report. 

Yes, the type promotion code wasn't properly handling this case. It also didn't handle the case where the type was a subclass of `int`:

```python
class IntSubclass(int): ...

def f(x: IntSubclass) -> float:
    return x
```

A fix for both issues will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.185, which I just published. It will also be included in the next release of pylance.
